### PR TITLE
Added back button to course detail page

### PR DESCRIPTION
**Pull request by realmayus**
Closes https://github.com/corona-school/project-user/issues/246

### CODE REVIEW

**Review by geroembser — COMMENTED** [review]
I don't think that the back button's destination is appropriate in all cases. 
Consider two cases: 
1. a user is logged in: The course details is then accessible under e.g. `/courses/6`, but then the back button will redirect to `/courses` in all cases. This is fine if a user has clicked on one of his courses, i.e. she originated from `/courses` route. But consider a user browsing the entire course library, i.e. she is originating from `/courses/overview`, then the actual destination of the back button should be `/courses/overview` to come back to the course overview.
2. No user is logged in (and thus accessing the public course page): The course details view then is available under e.g. `/public/courses/6`, but the back button will redirect to `/courses` which will be a protected route in a sense that it requires users to be logged in. That is not really intuitive. The actual destination should be `/public/courses`

And in general, it would be better to not push a new page on the browser's navigation stack but really go back – but since I know that this is way much harder to implement, it is ok to go with pushing to the history for now.

**Comment by LeonErath** [comment]
@geroembser why not use react routers `back` functionality which acts the same as the browsers back button?
```
<button onClick={() => {
        history.goBack();
 }}
>Go back</button>
```

**Comment by geroembser** [comment]
> @geroembser why not use react routers `back` functionality which acts the same as the browsers back button?
> 
> ```
> <button onClick={() => {
>         history.goBack();
>  }}
> >Go back</button>
> ```

@LeonErath if we would check if we can go back or not (i.e. check the browser navigation stack), I think that would be a good idea. I was thinking that this would not be the best solution because there are a lot of email campaigns sent to our users that link directly to the public details page of a course (say `/public/courses/6`). In most cases, when a user clicks on that button, nothing will happen if they just go back in the history. However, we want our users to be able to browse the courses. Probably some kind of state management is needed here, but I don't know to be honest what is a solution that is easy to implement and also works when linking directly to the details page.

**Comment by realmayus** [comment]
I think the best approach to this problem would be to store the current scroll position as an anchor tag in the URL, like this:
`https://localhost:5000/courses/overview#Naturwissenschaften:6`
Then I'd implement a function, that acts as the callback of the API call that gets all courses, that will read the anchor from the URL and scroll to the "Naturwissenschaften" section and go to the 6th item there. 
That way, we'd be able to get around Redux and whatnot.

**Review by lsc0498 — APPROVED** [review]
Looks good to me. Very nice! 🎉

File: src/routes/CourseOverview.tsx [hunk 1]
```diff
@@ -32,8 +39,58 @@ export const CourseOverview: React.FC<Props> = ({
...
+  const scrollToItemInSection = (hash, courseList, container) => {
+    const courseId = Number(decodeURIComponent(hash.slice(1).split(':')[1]));
+    if (courseId != null) {
+      setTimeout(() => {
+        console.log(courseId, courseList);
+        const course = courseList.find((c) => c.id === courseId);
+        if (course != null) {
+          // eslint-disable-next-line no-param-reassign
+          container.scrollLeft = course.el.getBoundingClientRect().x;
+          // scrollToTargetAdjustedRight(container, course.el, 0);
```
Comment (lsc0498) [inline src/routes/CourseOverview.tsx:56]: Can you remove these commented out lines before merging?

**Review by lsc0498 — APPROVED** [review]
Still looks good 😉 Thank you!

**Pull request merged**